javascript/treeIntersection/tree-intersection.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+class Node {
+    constructor(value) {
+      this.value = value;
+      this.left = null;
+      this.right = null;
+    }
+  }
+  
+  class BinarySearchTree {
+    constructor() {
+      this.root = null;
+    }
+  
+    insert(value) {
+      const newNode = new Node(value);
+  
+      if (!this.root) {
+        this.root = newNode;
+        return;
+      }
+  
+      let current = this.root;
+  
+      while (true) {
+        if (value === current.value) {
+          return;
+        }
+  
+        if (value < current.value) {
+          if (!current.left) {
+            current.left = newNode;
+            return;
+          }
+          current = current.left;
+        } else {
+          if (!current.right) {
+            current.right = newNode;
+            return;
+          }
+          current = current.right;
+        }
+      }
+    }
+  
+    // Helper function for tree traversal
+    _traverse(node, callback) {
+      if (node) {
+        this._traverse(node.left, callback);
+        callback(node.value);
+        this._traverse(node.right, callback);
+      }
+    }
+  
+    // Returns an array of values in the tree
+    traverse() {
+      const values = [];
+      this._traverse(this.root, value => values.push(value));
+      return values;
+    }
+  }
+  
+  function tree_intersection(tree1, tree2) {
+    const hashmap = new Map();
```

Comment:
You're not really storing anything with the value, so you could just use a Set here.